File: src/constants/studyFrequency.js
export const STUDY_FREQUENCIES = [
  {
    value: 'daily',
    label: 'Daily',
    sessionsPerWeek: 7,
    description: '7 sessions per week',
  },
  {
    value: 'twice_weekly',
    label: 'Twice a Week',
    sessionsPerWeek: 2,
    description: '2 sessions per week',
  },
  {
    value: 'four_times_weekly',
    label: '4 Times a Week',
    sessionsPerWeek: 4,
    description: '4 sessions per week',
  },
  {
    value: 'weekly',
    label: 'Weekly',
    sessionsPerWeek: 1,
    description: '1 session per week',
  },
];

const byValue = new Map(STUDY_FREQUENCIES.map(item => [item.value, item]));

export function studyFrequencyLabel(value) {
  if (!value) return 'Not Selected';
  return byValue.get(value)?.label ?? 'Not Selected';
}

export function isValidStudyFrequency(value) {
  return byValue.has(value);
}

File: backend/app/services/study_frequency.py
"""Study schedule preference values and display labels."""

from __future__ import annotations

STUDY_FREQUENCY_DAILY = "daily"
STUDY_FREQUENCY_TWICE_WEEKLY = "twice_weekly"
STUDY_FREQUENCY_FOUR_TIMES_WEEKLY = "four_times_weekly"
STUDY_FREQUENCY_WEEKLY = "weekly"

ALLOWED_STUDY_FREQUENCIES = frozenset(
    {
        STUDY_FREQUENCY_DAILY,
        STUDY_FREQUENCY_TWICE_WEEKLY,
        STUDY_FREQUENCY_FOUR_TIMES_WEEKLY,
        STUDY_FREQUENCY_WEEKLY,
    }
)

STUDY_FREQUENCY_LABELS: dict[str, str] = {
    STUDY_FREQUENCY_DAILY: "Daily",
    STUDY_FREQUENCY_TWICE_WEEKLY: "Twice a Week",
    STUDY_FREQUENCY_FOUR_TIMES_WEEKLY: "4 Times a Week",
    STUDY_FREQUENCY_WEEKLY: "Weekly",
}


def study_frequency_label(value: str | None) -> str:
    if value is None:
        return "Not Selected"
    return STUDY_FREQUENCY_LABELS.get(value, "Not Selected")


def validate_study_frequency(value: str) -> str:
    cleaned = value.strip()
    if cleaned not in ALLOWED_STUDY_FREQUENCIES:
        raise ValueError(
            "study_frequency must be one of: daily, twice_weekly, four_times_weekly, weekly"
        )
    return cleaned
